corehq/form_processor/parsers/form.py

```diff
@@ -147,12 +150,15 @@ def _handle_id_conflict(xform, domain):
     interface = FormProcessorInterface(domain)
     if interface.is_duplicate(conflict_id, domain):
         # It looks like a duplicate/edit in the same domain so pursue that workflow.
+        logger.info('Handling duplicate doc id %s for domain %s', conflict_id, domain)
         return _handle_duplicate(xform)
     else:
         # the same form was submitted to two domains, or a form was submitted with
         # an ID that belonged to a different doc type. these are likely developers
         # manually testing or broken API users. just resubmit with a generated ID.
-        return interface.assign_new_id(xform), None
+        interface.assign_new_id(xform)
```

Comment:
This changed slightly from previous behavior: the value returned by this method call used to be returned, but now is ignored. Looking at the implementations of `assign_new_id()` I see it does not matter since the object passed in is the object returned, but noting in case that was not an intentional change.

Reply:
Yeah, I looked into the method and made sure that I wasn't changing behavior, as the intention was only to record the different ids

Reply:
Not necessary in this PR, but it might be reasonable to refactor `assign_new_id()` to not return a value and change all the places where it's called to not use the return value.